sdks/go/pkg/beam/core/runtime/exec/sdf_invokers.go

```diff
@@ -351,3 +352,84 @@ func (n *cweInvoker) Reset() {
 		n.args[i] = nil
 	}
 }
+
+// trInvoker is an invoker for TruncateRestriction.
+type trInvoker struct {
+	fn   *funcx.Fn
+	args []interface{}
+	call func(elms *FullValue, rest interface{}) (pair interface{})
+}
+
+var offsetrangeTracker = reflect.TypeOf((*offsetrange.Tracker)(nil)).Elem()
+
+// var growableOffsetRangeTracker = reflect.TypeOf((*growable_offsetrange.Tracker)(nil))
+
+func DefaultTruncateRestriction(restTracker interface{}) (newRest interface{}) {
+	switch restTracker.(type) {
+	case *offsetrange.Tracker:
+		return restTracker.(*offsetrange.Tracker).GetRestriction().(offsetrange.Restriction) // since offsetrange has a bounded restriction
+	default:
+		return nil
```

Comment:
Yeah, it should be that generically. Since `offsetrange` has `IsBounded()` set to return `true` always, I wrote it that way. But it'll be changed to a generic way in the next commit.